src/Reflection/Adapter/ReflectionClass.php

```diff
@@ -49,7 +48,7 @@ public function __get(string $name): mixed
             return $this->betterReflectionClass->getName();
         }
 
-        throw new OutOfBoundsException(sprintf('Property %s::$%s does not exist.', self::class, $name));
+        return null;
```

Comment:
We should probably keep the `OutOfBoundsException` for dynamic properties

Reply:
I've tried to be compatible with the core reflection: https://3v4l.org/NlhUD

Reply:
Right, but here we just say `null` and all is good: core at least raises a notice :D 

This is something I tried replicating also in `ocramius/proxy-manager`, but on the long term, it's just a waste of time and a source of bugs: undefined properties are undefined, and we don't need to be compatible with core on it.

A hard crash or a notice are equivalent here: let's have a clear crash.

Reply:
Ok, the commit is away.